lib/decodeStateTransitionFactory.js

```diff
@@ -17,18 +15,13 @@ function decodeStateTransitionFactory(dpp) {
    * @return {DataContractStateTransition|DocumentsStateTransition}
    */
   async function decodeStateTransition(rawData) {
```

Comment:
As I remember we decided to move this logic to handlers and remove this function. It's kinda smelly coz domain logic shouldn't throw interface level exceptions. So now it's much smaller and it's ok to duplicate this code a bit in handers. Later we can introduce middleware for that.